src/main/java/com/project/portfolio/springboot/onlinesellingandshopping/serverapp/security/ApplicationUserDetailsService.java

```diff
@@ -0,0 +1,38 @@
+package com.project.portfolio.springboot.onlinesellingandshopping.serverapp.security;
+
+import com.project.portfolio.springboot.onlinesellingandshopping.serverapp.module.AuthGroup;
+import com.project.portfolio.springboot.onlinesellingandshopping.serverapp.module.User;
+import com.project.portfolio.springboot.onlinesellingandshopping.serverapp.repository.AuthGroupRepository;
+import com.project.portfolio.springboot.onlinesellingandshopping.serverapp.repository.UserRepository;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class ApplicationUserDetailsService implements UserDetailsService {
+
+    private final UserRepository userRepository;
+    private final AuthGroupRepository authGroupRepository;
+
+    public ApplicationUserDetailsService(UserRepository userRepository, AuthGroupRepository authGroupRepository) {
+        this.userRepository = userRepository;
+        this.authGroupRepository = authGroupRepository;
+    }
+
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = (User) this.userRepository.findByUsername(username).get();
+
+        if(null==user){
+            throw new UsernameNotFoundException("couldn't find username: " + username);
+        }
+
+        List<AuthGroup> authGroups = this.authGroupRepository.findByUsername(username);
```

Comment:
You can drop 'this' , any class member can be used in the class methods without explicitly using 'this' reference. ( unless there is a local variable with the same name ). Also it's best to add final when a local variable is defined and assigned value at the same time ( unless you intend to re-assign the value again later in the method block )